app/script/components/contextMenu.coffee

```diff
@@ -0,0 +1,101 @@
+#
+# Wire
+# Copyright (C) 2016 Wire Swiss GmbH
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see http://www.gnu.org/licenses/.
+#
+
+window.z ?= {}
+z.components ?= {}
+
+z.components.ContextMenuEvent =
+  CONTEXT_MENU: 'customContextMenu'
+
+# Display context menu for the given entries. Entries should be provided as an Array of object containing
+# 'label' and 'action'. Dom Event 'z.components.ContextMenuEvent.CONTEXT_MENU' is fired when the
+# context menu is clicked and data is updated.
+#
+# entries example:
+# [
+#   {label: 'copy', action: 'action-id'},
+#   {label: 'delete', action: 'action-id'}
+# ]
+#
+# @param [Object] params
+# @option params [String] tag
+# @option params [Object] data additional data that is passed then an items in the context menu is clicked
```

Comment:
`additional data that is passed then an items in the context menu is clicked`
->
`additional data that is passed when items in the context menu are clicked`